Makefile

```diff
@@ -25,34 +29,49 @@ $(INSTALL_STAMP): $(PYPROJECT_TOML) $(POETRY_LOCK)
 		echo "Poetry could not be found. See https://python-poetry.org/docs/"; \
 		exit 2; \
 	fi
-	$(POETRY) install --no-root --with circleci_scraper,dev
+	$(POETRY) install --no-root --with circleci_scraper,test_metric_reporter,dev
 	# Create an empty install stamp file to indicate that dependencies have been installed
 	touch $(INSTALL_STAMP)
 
 .PHONY: check
 check: $(INSTALL_STAMP)
-	# Run ruff to check for linting issues in the scripts directory
-	$(POETRY) run ruff check $(SCRIPT_DIR)
-	# Run ruff to check the formatting
-	$(POETRY) run ruff format --check $(SCRIPT_DIR)
+	# Run ruff to check for linting issues in the scripts and tests directories
+	$(POETRY) run ruff check $(CHECK_DIRS)
+	# Run ruff to check the formatting in the scripts and tests directories
+	$(POETRY) run ruff format --check $(CHECK_DIRS)
 	# Run bandit to check for security issues
-	$(POETRY) run bandit --quiet -r -c $(PYPROJECT_TOML) $(SCRIPT_DIR)
+	$(POETRY) run bandit --quiet -r -c $(PYPROJECT_TOML) $(CHECK_DIRS)
 	# Run mypy to check for type issues
-	$(POETRY) run mypy --config-file=$(PYPROJECT_TOML) $(SCRIPT_DIR)
+	$(POETRY) run mypy --config-file=$(PYPROJECT_TOML) $(CHECK_DIRS)
 
 .PHONY: format
 format: $(INSTALL_STAMP)
-	# Run ruff to automatically fix linting issues
-	$(POETRY) run ruff check --fix $(SCRIPT_DIR)
-	# Run ruff to format the scripts
-	$(POETRY) run ruff format $(SCRIPT_DIR)
+	# Run ruff to automatically fix linting issues in the scripts and tests directories
+	$(POETRY) run ruff check --fix $(CHECK_DIRS)
+	# Run ruff to format the scripts and tests directories
+	$(POETRY) run ruff format $(CHECK_DIRS)
+
+.PHONY: test
+test: $(INSTALL_STAMP)
+	$(POETRY) run pytest --cov=$(SCRIPTS_DIR)
+
+.PHONY: test-coverage-html
+test-coverage-html: $(INSTALL_STAMP)
+	$(POETRY) run pytest --cov=$(SCRIPTS_DIR) --cov-report=html
 
 .PHONY: clean
 clean:
 	rm -f $(INSTALL_STAMP)
+	rm -f .coverage
 	rm -rf .mypy_cache
 	rm -rf .ruff_cache
+	rm -rf .pytest_cache
+	rm -rf htmlcov
 
 .PHONY: run_circleci_scraper
```

Comment:
Getting an error on this target now (possibly w/ the config file). Investigating...:
```sh
make run_circleci_scraper

PYTHONPATH=. poetry run python scripts/circleci_scraper/main.py --config=config.ini
2024-07-15 10:16:32,196 - ERROR - The 'common' section is missing
Traceback (most recent call last):
  File "/Volumes/Dev/github/mozilla/ecosystem-test-scripts/scripts/common/config.py", line 53, in _parse_main_config
    test_result_dir=config_parser.get("common", "test_result_dir"),
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.12/3.12.3/Frameworks/Python.framework/Versions/3.12/lib/python3.12/configparser.py", line 759, in get
    d = self._unify_values(section, vars)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/Cellar/python@3.12/3.12.3/Frameworks/Python.framework/Versions/3.12/lib/python3.12/configparser.py", line 1132, in _unify_values
    raise NoSectionError(section) from None
configparser.NoSectionError: No section: 'common'
2024-07-15 10:16:32,197 - ERROR - Configuration error: The 'common' section is missing (caused by No section: 'common')
```

Reply:
Not sure if you've possibly edited your local config.ini but possibly not back ported the changes to the config.ini.sample?

Or maybe you renamed `[main]` to `[common]` locally?
```ini
[main]
test_result_dir = raw_data
test_metadata_dir = circle_ci
test_artifact_dir = junit
```